app/apollo/resolvers/channel.js

```diff
@@ -136,6 +135,25 @@ const channelResolvers = {
         // search version by version uuid or version name
         const versionObj = channel.versions.find(v => (v.uuid === versionUuid || v.name === versionName));
         if (!versionObj) {
+          const orphanedDeployableVersionObj = await models.DeployableVersion.findOne({org_id, channel_id: channel_uuid, $or: [{uuid: versionUuid},{name: versionName}] });
+          if (orphanedDeployableVersionObj) {
+            logger.warn(
+              {
+                org_id,
+                channel_uuid: channel.uuid,
+                channel_name: channel.name,
+                ver_uuid: versionUuid,
+                ver_name: versionName,
+                orphan_uuid: orphanedDeployableVersionObj.uuid,
+                orphan_name: orphanedDeployableVersionObj.name,
+                orphan_channel_id: orphanedDeployableVersionObj.channel_id,
+                orphan_channel_name: orphanedDeployableVersionObj.channel_name,
+                queryName
+              },
+              'possible orphan -- Version found with matching name or uuid, but it is not referenced by the Channel'
```

Comment:
I consider this useless code, which just adds noise and increases maintenance cost: more code -> more errors -> more effort to maintain.

Reply:
The code is not useless, as it provides additional visibility when orphans occur (and we know they are possible).  With this section, it's possible to use the logs to determine the UUID of the orphaned Version to allow manual deletion via `removeChannelVersion` API.  Without this section there would be no indication of _why_ the API is claiming the Version does not exist, and no other way to determine the UUID when searching by Version name.

Reply:
I agree, it may be useful for troubleshooting problems, but this kind of "institutionalizes" the data corruption problem in the code.

Reply:
Removed